content/en/blog/2022/frontend-overhaul/index.md

```diff
@@ -0,0 +1,182 @@
+operating system your team is using, as well as providing different approaching
+techniques (automatic and manual instrumentation, metrics, baggage). All of this
+while following the standards and conventions defined by the official
+OpenTelemetry Documentation. More about the specific requirements can be
+[found here](https://github.com/open-telemetry/opentelemetry-demo/tree/main/docs/requirements).
+
+My company was focused on becoming part of and embracing the OpenTelemetry
+community. One of our goals this summer was to get more involved with a core
+OpenTelemetry project where we could provide a meaningful contribution. The OTel
+demo became the best match for achieving that goal as it would not only help the
+community, but help our product as we needed a good example to test and showcase
+what can be done with our tool.
+
+The first thing we did was to get in contact with Carter Socha, the organizer of
+the OTel Demo SIG. Carter was really welcoming and helped us identify where our
+contributions could be the most impactful. We started looking at
+[the issue created by Austin Parker](https://github.com/open-telemetry/opentelemetry-demo/issues/39)
+referencing a complete front-end overhaul that would involve moving the
+application away from Go (SSR) to an architecture that included a browser-side
+client (CSR), as well as improving the overall style, theme, and user
+experience.
+
+A fun aspect of the work was the request to move the store from a ‘normal’ store
+to an astronomy store to match the OpenTelemetry project’s overall branding.
+
+Once we got confirmation and the green light from the rest of the OTel Demo SIG
+team then my team started working on the different changes included as part of
+the application frontend architecture overhaul.
+
+## OpenTelemetry Demo Application Description and Tech Stack
+
+The demo app is an astronomy webstore that has the basic functionality to
+purchase online products such as a shopping cart, currency selector, product
+listing, and payment & checkout. It also includes features to display
+promotional items (ads) and related products depending on the context of the
+user. The demo stack includes a set of multiple microservices built in different
+languages, one for each of the following programming languages:
+
+1. Go
+2. PHP
+3. C++
+4. .NET
+5. Ruby
+6. Python
+7. Node.js
+8. Rust
+9. Elixir
+
+Every microservice has a specific goal and can communicate with others by using
+a global GRPC definition. Persistent information is saved into a Postgres
+database and there are outbound services that connect with third-party services
+to trigger events (such as confirmation emails). All of the microservices,
+including the front end, are connected to the same Open Telemetry collector
+instance, which uses Jaeger as one of the data stores for the traces and spans.
+
+![Open Telemetry Demo System Diagram](system-diagram.png)
+
+The front end was constituted by a Golang SSR application which sent the
```

Comment:
Rewording:
“Prior to re-architecting, the front end consisted of a Golang SSR app, which sent complete HTML to the browser for display. Every request and call redirected to the server so new information was shown.”